### PR TITLE
pass close event to reducer, add reconnect callback

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by notgiorgi** [comment]
Hi, thanks for the PR, can you run the linter on the code?

**Review by notgiorgi — CHANGES_REQUESTED** [review]
I think you formatted the code with different linting rules, please run `npm run lint`

Also I don't really like the idea of passing that callback everywhere as argument, can you think of any other approaches?

File: src/actions.js [hunk 1]
```diff
@@ -12,9 +12,10 @@ export function createConnectionAction (endpoint) {
   }
 }
 
-export function createDisonnectionAction (endpoint) {
+export function createDisonnectionAction (endpoint, clode) {
```
Comment (notgiorgi) [inline src/actions.js]: arg name is incorrect

File: src/lib/Connection.js [hunk 2]
```diff
@@ -3,10 +3,19 @@ import MessageQueue from './MessageQueue'
 import { JSONCodec } from './Codecs'
 
 export default class Connection {
-  constructor (endpoint, queue = new MessageQueue(), codec = JSONCodec, Socket = window.WebSocket) {
+  constructor (
+    endpoint,
+    store,
+    reconnectCallback = null,
```
Comment (notgiorgi) [inline src/lib/Connection.js]: add this as a last argument

File: src/lib/Connection.js [hunk 3]
```diff
@@ -61,19 +72,22 @@ export default class Connection {
...
   _onError (err) {
     this.handlers.onError(err)
     this._startBackingOff()
   }
 
-  _onClose () {
-    this.handlers.onClose()
-    this._startBackingOff()
+  _onClose (close) {
+    console.log('_onClose')
```
Comment (notgiorgi) [inline src/lib/Connection.js]: remove log

**Comment by ghost** [comment]
hi, changes are done

**Comment by erudisch** [comment]
Hi, Sadly I've got no other idea other than passing the reconnectCallback via the ConnectionManager to the Connection...

**Review by ghost — COMMENTED** [review]
Hi, Sadly I've got no other idea other than passing the reconnectCallback via the ConnectionManager to the Connection...

**Review by ghost — COMMENTED** [review]
ok, what do you think to pass the callback to the Connection via the add method in createWebsocketMiddleware()?

File: src/actions.js (hunk 1 above)
Comment (ghost) [reply src/actions.js]: done

File: src/lib/Connection.js (hunk 2 above)
Comment (ghost) [reply src/lib/Connection.js]: done

File: src/lib/Connection.js (hunk 3 above)
Comment (ghost) [reply src/lib/Connection.js]: done

**Review by ghost — COMMENTED** [review]
Hi, could you please tell me what else i have to do that you accept the pull request?

Regards erik

**Review by notgiorgi — APPROVED** [review]

**Comment by ghost** [comment]
hi, when do you publish it at npm?

**Pull request merged**